rust/macros/syn/mod.rs

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT or Apache-2.0
+
+//! Shared parsing functions for use in procedural macros. These functions are
+//! from [syn](https://github.com/dtolnay/syn).
```

Comment:
```suggestion
//! Shared parsing functions for use in procedural macros.
//!
//! These functions come from the [`syn`](https://github.com/dtolnay/syn) crate.
```

Are the files too different from the original ones? If not, perhaps recording in the comment the Git tag from where they were checked out would be nice. If yes, doing it in the commit message is fair.

Reply:
I does make a few changes to strip away anything that depends on macros, bigint and unicode-xid. But I agree it's better to also record version number in the comment.

Reply:
That's great, thanks for removing those bits.

Reply:
Also mentioning that in the comment would be nice too (i.e. so that people doing future updates, if any, are aware of that and do not add `bigint` etc. back).